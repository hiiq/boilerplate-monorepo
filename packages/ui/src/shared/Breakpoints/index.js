import { useMediaQuery } from 'react-responsive';

const Breakpoint = {
  HUGE: 1440,
  LARGE: 1170,
  MEDIUM: 768,
  SMALL: 450,
};

const Huge = ({ children }) => {
  const isHuge = useMediaQuery({ minWidth: Breakpoint.HUGE });

  return isHuge ? children : null;
};

const Desktop = ({ children }) => {
  const isDesktop = useMediaQuery({ minWidth: Breakpoint.LARGE });

  return isDesktop ? children : null;
};

const Tablet = ({ children }) => {
  const isTablet = useMediaQuery({ minWidth: Breakpoint.MEDIUM });

  return isTablet ? children : null;
};

const Mobile = ({ children }) => {
  const isMobile = useMediaQuery({ maxWidth: Breakpoint.SMALL - 1 });

  return isMobile ? children : null;
};

const GreaterThanMobile = ({ children }) => {
  const isMobile = useMediaQuery({ minWidth: Breakpoint.SMALL });

  return isMobile ? children : null;
};

const Default = ({ children }) => {
  const isNotMobile = useMediaQuery({ minWidth: 0 });

  return isNotMobile ? children : null;
};

export { Default, Desktop, GreaterThanMobile, Huge, Mobile, Tablet };
