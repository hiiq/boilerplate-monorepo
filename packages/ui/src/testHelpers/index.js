import { cleanup, render } from '@testing-library/react';
import { curry } from 'ramda';
import React from 'react';
import { Providers } from 'modules/app';

export * from '@testing-library/react';

const customRender = curry((Component, defaultProps = {}, overrides = {}) =>
  render(<Component {...defaultProps} {...overrides} />, {
    wrapper: Providers,
  })
);

export { cleanup, customRender as render };
