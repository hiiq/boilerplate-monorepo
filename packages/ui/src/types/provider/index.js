import { Context } from '../context';

const Provider = {
  AUTH: Context.AUTH.Provider,
  FEATURE_FLAGS: Context.FEATURE_FLAGS.Provider,
  INTERNET_CONNECTIVITY: Context.INTERNET_CONNECTIVITY.Provider,
  MODALITY: Context.MODALITY.Provider,
  THEME_SWITCHER: Context.THEME_SWITCHER.Provider,
};

export { Provider };
