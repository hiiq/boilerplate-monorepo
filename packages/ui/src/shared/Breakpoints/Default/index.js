import { useMediaQuery } from 'shared/useMediaQuery';

const Default = ({ children }) => {
  const isNotMobile = useMediaQuery({ minWidth: 0 });

  return isNotMobile ? children : null;
};

export { Default };
