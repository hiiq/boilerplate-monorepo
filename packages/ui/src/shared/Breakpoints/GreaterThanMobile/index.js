import { Breakpoint } from '@boilerplate-monorepo/ui-common';
import { useMediaQuery } from 'shared/useMediaQuery';

const GreaterThanMobile = ({ children }) => {
  const isMobile = useMediaQuery({ minWidth: Breakpoint.SMALL });

  return isMobile ? children : null;
};

export { GreaterThanMobile };
