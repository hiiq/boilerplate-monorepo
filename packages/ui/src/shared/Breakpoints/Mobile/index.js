import { Breakpoint } from '@boilerplate-monorepo/ui-common';
import { useMediaQuery } from 'shared/useMediaQuery';

const Mobile = ({ children }) => {
  const isMobile = useMediaQuery({ maxWidth: Breakpoint.SMALL - 1 });

  return isMobile ? children : null;
};

export { Mobile };
