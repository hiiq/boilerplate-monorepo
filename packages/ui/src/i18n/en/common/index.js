import { Site } from 'types/site';

export const common = {
  about: 'About',
  dashboard: 'Dashboard',
  footer: `©️ {{fullYear}} ${Site.name}. All Rights reserved.`,
  home: 'Home',
  loading: 'Loading',
  loadingScreenReader: 'Please wait while content is being loaded',
  siteName: `${Site.name} :: {{title}}`,
};
