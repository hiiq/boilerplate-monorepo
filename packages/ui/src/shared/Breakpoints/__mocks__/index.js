import { Desktop } from '../Desktop';
import { GreaterThanMobile } from '../GreaterThanMobile';
import { Huge } from '../Huge';
import { Mobile } from '../Mobile';
import { Tablet } from '../Tablet';

const Breakpoints = {
  Desktop,
  GreaterThanMobile,
  Huge,
  Mobile,
  Tablet,
};

export { Breakpoints };
