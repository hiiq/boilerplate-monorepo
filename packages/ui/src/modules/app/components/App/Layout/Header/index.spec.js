import { User } from '@boilerplate-monorepo/common';
import React from 'react';
import * as GreaterThanMobileBreakpoint from 'shared/Breakpoints/GreaterThanMobile';
import * as MobileBreakpoint from 'shared/Breakpoints/Mobile';
import { render } from 'testHelpers';
import { Provider } from 'types/provider';
import { Header } from '.';

jest.mock('shared/useAuth');
jest.mock('./UserMenu', () => ({
  UserMenu: (props) => <x-UserMenu {...props} />,
}));
jest.mock('./SkipToContentLink', () => ({
  SkipToContentLink: (props) => <x-SkipToContentLink {...props} />,
}));

describe('<Header/>', () => {
  const defaultProps = {};

  const baseAuthContext = {
    login: jest.fn().mockName('login'),
    logout: jest.fn().mockName('logout'),
    user: User.uiExample({
      email: 'EMAIL',
    }),
  };

  const renderComponent = (authContext) => (overrides) => {
    // eslint-disable-next-line react/prop-types
    const wrapper = ({ children }) => (
      <Provider.AUTH value={authContext}>{children}</Provider.AUTH>
    );

    return render(<Header {...defaultProps} {...overrides} />, { wrapper });
  };

  describe('when authenticated', () => {
    const authContext = {
      ...baseAuthContext,
      isAuthenticated: true,
    };

    describe('when on mobile', () => {
      test('renders properly', () => {
        td.replace(
          GreaterThanMobileBreakpoint,
          'GreaterThanMobile',
          () => null
        );
        td.replace(MobileBreakpoint, 'Mobile', (props) => (
          <x-Mobile {...props} />
        ));

        const { container } = renderComponent(authContext)();

        expect(container.firstChild).toMatchSnapshot();
      });
    });

    describe('when on larger than mobile', () => {
      test('renders properly', () => {
        td.replace(
          GreaterThanMobileBreakpoint,
          'GreaterThanMobile',
          (props) => <x-GreaterThanMobile {...props} />
        );
        td.replace(MobileBreakpoint, 'Mobile', () => null);

        const { container } = renderComponent(authContext)();

        expect(container.firstChild).toMatchSnapshot();
      });
    });
  });

  describe('when not authenticated', () => {
    const authContext = {
      ...baseAuthContext,
      isAuthenticated: false,
    };

    describe('when on mobile', () => {
      test('renders properly', () => {
        td.replace(
          GreaterThanMobileBreakpoint,
          'GreaterThanMobile',
          () => null
        );
        td.replace(MobileBreakpoint, 'Mobile', (props) => (
          <x-Mobile {...props} />
        ));

        const { container } = renderComponent(authContext)();

        expect(container.firstChild).toMatchSnapshot();
      });
    });

    describe('when on larger than mobile', () => {
      test('renders properly', () => {
        td.replace(
          GreaterThanMobileBreakpoint,
          'GreaterThanMobile',
          (props) => <x-GreaterThanMobile {...props} />
        );
        td.replace(MobileBreakpoint, 'Mobile', () => null);

        const { container } = renderComponent(authContext)();

        expect(container.firstChild).toMatchSnapshot();
      });
    });
  });
});
