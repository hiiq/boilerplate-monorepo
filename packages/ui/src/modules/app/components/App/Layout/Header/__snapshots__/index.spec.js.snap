// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> when authenticated when on larger than mobile renders properly 1`] = `
<header
  class="Header__StyledHeader-sc-4wz997-1 cPWXaN"
  role="banner"
  title="Acme Inc"
>
  <x-skiptocontentlink />
  <div
    class="Header__Container-sc-4wz997-0 kkfqiI"
  >
    <div
      class="EllipsiedText-o83sq0-0 bVGnli"
    >
      Acme Inc
    </div>
    <x-greaterthanmobile>
      <button
        aria-disabled="false"
        aria-label="t(logOut)"
        class="Button__Container-sc-1v65cga-0 ggxdnx Header__ImageButton-sc-4wz997-2 eSaMoc"
        role="button"
        type="button"
      >
        <x-avatar />
      </button>
    </x-greaterthanmobile>
  </div>
</header>
`;

exports[`<Header/> when authenticated when on mobile renders properly 1`] = `
<header
  class="Header__StyledHeader-sc-4wz997-1 cPWXaN"
  role="banner"
  title="Acme Inc"
>
  <x-skiptocontentlink />
  <div
    class="Header__Container-sc-4wz997-0 kkfqiI"
  >
    <div
      class="EllipsiedText-o83sq0-0 bVGnli"
    >
      Acme Inc
    </div>
    <x-mobile>
      <x-navigation />
    </x-mobile>
  </div>
</header>
`;

exports[`<Header/> when not authenticated when on larger than mobile renders properly 1`] = `
<header
  class="Header__StyledHeader-sc-4wz997-1 cPWXaN"
  role="banner"
  title="Acme Inc"
>
  <x-skiptocontentlink />
  <div
    class="Header__Container-sc-4wz997-0 kkfqiI"
  >
    <div
      class="EllipsiedText-o83sq0-0 bVGnli"
    >
      Acme Inc
    </div>
    <x-greaterthanmobile>
      <button
        aria-disabled="false"
        aria-label="t(logOut)"
        class="Button__Container-sc-1v65cga-0 ggxdnx Header__ImageButton-sc-4wz997-2 eSaMoc"
        role="button"
        type="button"
      >
        <x-avatar />
      </button>
    </x-greaterthanmobile>
  </div>
</header>
`;

exports[`<Header/> when not authenticated when on mobile renders properly 1`] = `
<header
  class="Header__StyledHeader-sc-4wz997-1 cPWXaN"
  role="banner"
  title="Acme Inc"
>
  <x-skiptocontentlink />
  <div
    class="Header__Container-sc-4wz997-0 kkfqiI"
  >
    <div
      class="EllipsiedText-o83sq0-0 bVGnli"
    >
      Acme Inc
    </div>
    <x-mobile>
      <x-navigation />
    </x-mobile>
  </div>
</header>
`;
